client/account-recovery/reset-code-validation/index.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { get } from 'lodash';
+import qs from 'qs';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+
+import {
+	updatePasswordResetUserData,
+	setResetMethod,
+	validateRequest,
+	validateRequestError,
+} from 'state/account-recovery/reset/actions';
+
+import {
+	getAccountRecoveryValidationError,
+	getAccountRecoveryValidationKey,
+} from 'state/selectors';
+
+class ResetPasswordEmailValidation extends Component {
+	getQueryString = () => {
+		const queryString = get( window, 'location.search', null );
+
+		if ( ! queryString ) {
+			return '';
+		}
+
+		return queryString.slice( 1 );
+	}
+
+	parseQueryArgs = () => ( qs.parse( this.getQueryString() ) );
+
+	componentDidMount = () => {
+		const {
+			user,
+			method,
+			key,
+		} = this.parseQueryArgs();
+
+		if ( ! user || ! method || ! key ) {
+			this.props.validateRequestError( {
+				message: this.props.translate( 'The validation URL is incomplete.' ),
+			} );
+
+			return;
+		}
+
+		this.props.validateRequest( { user }, method, key );
+	}
+
+	componentWillReceiveProps = ( nextProps ) => {
+		if ( ! this.props.validationKey && nextProps.validationKey ) {
+			const {
+				user,
+				method,
+			} = this.parseQueryArgs();
+
+			// Fill in the user & method field only if the validation is succeeded.
+			// <AccountRecoveryRoot> component is using redux state to determine which AR step we're at.
+			// Thus, we need to confirm if these data are correct before filling in, otherwise users would
+			// arrive at wrong steps.
+			this.props.updatePasswordResetUserData( { user } );
+			this.props.setResetMethod( method );
+		}
+	}
+
+	renderValidating = () => (
+		<p className="reset-code-validation__text-validating">{ this.props.translate( 'Validating' ) }</p>
+	)
+
+	renderErrorMsg = ( error ) => (
+		<p className="reset-code-validation__text-error">
+			{ error.message ? this.props.translate( "We've encountered an error: " ) + error.message
```

Comment:
Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Autosave encountered an unexpected error' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).